### ISSUE TITLE
subsequent runs of testStaticOptimization fail

### ISSUE THREAD

**Issue by aseth1** [opensim-org/opensim-core#647]
@aymanhab @chrisdembia 
A clean build and run of testStaticOptimization passes, but subequent runs fail.
I have verified that the `test/ResultsSO_spring/walk_subject01_ankle_spring_StaticOptimization_activation.sto` values change from the initial run to the second. 

Anyone else having this issue?

**Comment by aymanhab** [comment]
@aseth1 I don't have an environment to test this with but I'd conjecture such a problem would be caused by some test case(s) that share the same folder and data/setup files that end up modifying the input files and hence the test case producing different output. So I'd say very likely related to #614. Can you check if the input files are the same as the source tree?

**Comment by aseth1** [comment]
> I don't have an environment to test this with but I'd conjecture such a problem would be caused by some test case(s) that share the same folder and data/setup files that end up modifying the input files

Ayman, you never had to run the test twice (without a clean build between)? I quite agree that an input is likely being corrupted.

**Comment by aymanhab** [comment]
@aseth1 I ran the test likely tens of times but I'd always run the test by itself and in the source tree because running the test from VisualStudio doesn't automatically copy/refresh the files in the test directory.

**Comment by aseth1** [comment]
OK. That is strange. I get the failure just running `testStaticOptimization` in isolation a second time. If i refresh the test directory the test passes. That suggests to me that running `testStaticOptimization` is itself the culprit in changing the input files.

**Comment by aseth1** [comment]
It is `externalForces.xml` that is being mucked with.

Before running the test:
```XML
<lowpass_cutoff_frequency_for_load_kinematics>6</lowpass_cutoff_frequency_for_load_kinematics>
```
After:
```XML
<lowpass_cutoff_frequency_for_load_kinematics> -1</lowpass_cutoff_frequency_for_load_kinematics>
```
I recall seeing this change in your initial PR. Any ideas what is making this change automatically?

**Comment by aseth1** [comment]
My guess is a bug in `updateFromXML` for the (Analyze) Tool.

**Comment by aymanhab** [comment]
I'd agree that's a likely culprit, I vaguely remember your comment about the filtering change and that I undid it but it has been a long while. I'll investigate in my own environment.

**Comment by aymanhab** [comment]
Will be addressed by #652 Caused by missing header (Document version number)  in setup file that caused wrong conversion code to be invoked, a bug that was fixed in version 2.0. All released setup files that I checked have version numbers on them, but would be good to have another pair of eyes check that.
